nx/lib/nx.ex

```diff
@@ -6302,14 +6302,21 @@ defmodule Nx do
         10
       >
 
-  If a tensor of floats is given, it still returns integers:
+  If a tensor of floats or complex numbers is given, it still returns integers:
 
       iex> Nx.argmax(Nx.tensor([2.0, 4.0]))
       #Nx.Tensor<
         s64
         1
       >
 
+     iex> Nx.argmax(Nx.tensor([Complex.new(1, 2), Complex.new(3,2)]))
```

Comment:
```suggestion
     iex> Nx.argmax(Nx.tensor([Complex.new(1, 2), Complex.new(3,0)]))
```

I'm curious on what this would return. Also, if this is considered a tie

Reply:
On second thoughts, this should actually error out, as we can't really compare 2 complex numbers.

Reply:
Jax and Pytorch have a defined behavior for this. Perhaps ensure that we have the same behavior